### ISSUE TITLE
Add incorrect version warning to copy-frameworks

### ISSUE THREAD

**Issue by Zyphrax** [Carthage/Carthage#2101]
When you compile your project CocoaPods checks that the `Manifest.lock` file matches the `Podfile.lock` file to ensure that the local Pods are matching the ones required by the project.

It would be great if Carthage could do a similar check, to ensure that the Carthage/Build folder contains the frameworks that match the `Cartfile.resolved` file.

Perhaps it would be an idea to add this step to the `copy-frameworks` command, so that Xcode will raise a warning when the situation isn't in sync.

When you work on a project with multiple individuals you can get some unpredictable results when some are using older versions of the frameworks.

**Comment by erichoracek** [comment]
Hi @Zyphrax! Please take a look at the [`carthage-verify`](https://github.com/Carthage/workflows/#carthage-verify) script and the discussion in #1839 and see if that could work for you to address this issue.

**Comment by Zyphrax** [comment]
@erichoracek Thank you, I'll check it out!

It is unfortunate that this isn't part yet of Carthage. I'm glad that I made the switch from CocoaPods to Carthage, but sometimes I do miss the convenience that everything goes near automatic with CocoaPods.

CocoaPods is like a kitchen machine, one big device that can do a lot, no one really knows how, when it breaks down and you try to fix it, it turns out that all the components are glued together and can't be dismantled.

Carthage on the other hand is your dad's expensive chisel. Focussed to do one thing well, but you need quite a lot of other tools to get the job done. Without proper instructions people cut themselves or decide to use it to open a paint can.

Basically I'm looking for a midway screwdriver :smile:
